test/sentry/integrations/oban/cron_test.exs

```diff
@@ -4,8 +4,16 @@ defmodule Sentry.Integrations.Oban.CronTest do
 
   import Sentry.TestHelpers
 
-  setup_all do
-    Sentry.Integrations.Oban.Cron.attach_telemetry_handler()
+  setup context do
+    opts =
+      if context[:custom_monitor_name_generator] do
+        [monitor_name_generator: {__MODULE__, :custom_name_generator, []}]
+      else
+        []
+      end
```

Comment:
You can do

```suggestion
    opts = context[:attach_opts]
```

and then below define the test as

```elixir
@tag attach_opts: [monitor_name_generator: {__MODULE__, :custom_name_generator}]
```

It should be a bit easier to follow and require less code 🙃